### PR TITLE
make-disk-image: replace nixos-install by nix-env calls

### PR DESCRIPTION
**Pull request by ck3d**
nixos-install can only be called from a machine

###### Motivation for this change
fixes ova build error, which was introduced with PR #67232 .

###### Things done
I build ova successfully with following call:
```
nix-build ./nixos/release.nix -A ova.x86_64-linux
```

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @vcunat @fpletz @markuskowa

### CODE REVIEW

File: nixos/lib/make-disk-image.nix [hunk 1]
```diff
@@ -180,9 +180,17 @@ let format' = format; in let
     export NIX_STATE_DIR=$TMPDIR/state
     nix-store --load-db < ${closureInfo}/registration
 
-    echo "running nixos-install..."
-    nixos-install --root $root --no-bootloader --no-root-passwd \
-      --system ${config.system.build.toplevel} --channel ${channelSources} --substituters ""
+    mkdir -m 0755 -p "$root/etc"
+    touch "$root/etc/NIXOS"
```
Comment (lovesegfault) [inline nixos/lib/make-disk-image.nix]: Why is `NIXOS` in allcaps?

Comment (markuskowa) [reply nixos/lib/make-disk-image.nix]: That file indicates that is a NixOS system (see for example https://nixos.org/nixos/manual/index.html#sec-installing-from-other-distro).

Comment (flokli) [reply nixos/lib/make-disk-image.nix]: We might want to drop that file in the future, and rely on `/etc/os-release`: http://0pointer.de/blog/projects/os-release cc @arianvp @aanderse

**Comment by markuskowa** [comment]
The ova is working again. Also the following tests pass:
* `nixosTests.virtualbox`
* `amazonImage.x86_64-linux` (release.nix)
However, `nixosTests.ec2` and `nixosTests.openstack-image` are still broken (not sure this is related to the change in `nixos-install`).

**Review by markuskowa — APPROVED** [review]

**Comment by edolstra** [comment]
Not in favor of this. The whole point of calling `nixos-install` is to ensure we produce a consistent result.

**Comment by edolstra** [comment]
Reverted in 39c0e489d3e3e56ae848075bb6b2257176d48d32.

**Comment by arianvp** [comment]
@edolstra should we revert https://github.com/NixOS/nixpkgs/pull/67232/files for the time being too then? until we figure out why it broke `nixos-install` ?

**Comment by worldofpeace** [comment]
@arianvp Yes please. I understand someone can bring about the optimal solution, but there's no reason to block things because of this.

**Pull request merged**